### PR TITLE
Add test to keep needed fields public

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
As discussed in #16.

### CODE REVIEW

File: tests/public.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+extern crate mp4parse as mp4;
+
+use std::io::Cursor;
+
+// Took from https://github.com/GuillaumeGomez/audio-video-metadata/blob/9dff40f565af71d5502e03a2e78ae63df95cfd40/src/metadata.rs#L53
+//
+// Not important if it fails or not. If it compiles, it means it's all good.
+#[test]
+fn check_mp4() {
+    let mut c = Cursor::new(b"\x00\x00\x00\x13\x66\x74\x79\x70\x30\x30\x30\x30\x30\x30\x30\x30".to_vec());
```
Comment (rillian) [inline tests/public.rs]: Theres examples/minimal.mp4 you can test against too.

File: tests/public.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+extern crate mp4parse as mp4;
+
+use std::io::Cursor;
+
+// Took from https://github.com/GuillaumeGomez/audio-video-metadata/blob/9dff40f565af71d5502e03a2e78ae63df95cfd40/src/metadata.rs#L53
+//
+// Not important if it fails or not. If it compiles, it means it's all good.
```
Comment (rillian) [inline tests/public.rs]: Surely it should run to completion? :)

File: tests/public.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline tests/public.rs]: Even better, thanks!

**Comment by rillian** [comment]
Thanks. This is a good start, but you marked a lot more fields public. Maybe see how many can be checked with the in-tree test file?

**Comment by GuillaumeGomez** [comment]
Making it more global.

**Comment by GuillaumeGomez** [comment]
Ok, so now we have something even more complete. Also, I added myself as an author, I intend to contribute in here in the future (work linked to my crate). If you don't want it, I'll just remove the commit.

File: tests/public.rs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+fn check_mp4() {
+    let buf = if let Some(mut fd) = File::open("examples/minimal.mp4").ok() {
+        let mut buf = Vec::new();
+
+        match fd.read_to_end(&mut buf) {
+            Ok(_) => buf,
+            Err(_) => panic!("File error"),
+        }
+    } else {
+        panic!("Unknown file")
```
Comment (rillian) [inline tests/public.rs]: I'd rather have this up near the `File::open` and save a level of indent. For example you could use a pair of `match` stanzas like in `examples/dump.rs`. I think the readability is more important than freeing the file object sooner.

Using .expect() would be appropriate in a test too.

File: tests/public.rs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+extern crate mp4parse as mp4;
+
+use std::io::{Cursor, Read};
+use std::fs::File;
+
+// Took from https://github.com/GuillaumeGomez/audio-video-metadata/blob/9dff40f565af71d5502e03a2e78ae63df95cfd40/src/metadata.rs#L53
+//
+// Not important if it fails or not. If it compiles, it means it's all good.
```
Comment (rillian) [inline tests/public.rs]: Do these comments still apply?

File: tests/public.rs [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                        }
+                    }, "ES");
+                    assert!(a.samplesize > 0);
+                    assert!(a.samplerate > 0);
+                }
+                Some(mp4::SampleEntry::Unknown) | None => {}
+            }
+        }
+    } else {
+        panic!("read_mp4 function failed")
```
Comment (rillian) [inline tests/public.rs]: Same here, but more so. `if let Ok(())` is cute but the `else` clause is far removed from its conditional. Early returns are better.

**Comment by rillian** [comment]
Thanks for adding a more comprehensive test! Please address the comments and update the PR.

File: tests/public.rs (hunk 4 above)
Comment (GuillaumeGomez) [inline tests/public.rs]: Ah damn, no. I forgot to remove it.

File: tests/public.rs (hunk 3 above)
Comment (GuillaumeGomez) [inline tests/public.rs]: For me, the readability is at its best here. But if you prefer another match, it's fine by me.

File: tests/public.rs (hunk 5 above)
Comment (GuillaumeGomez) [inline tests/public.rs]: Ok!

**Comment by GuillaumeGomez** [comment]
Updated.

File: Cargo.toml [hunk 6]
```diff
@@ -4,6 +4,7 @@ version = "0.4.1"
 authors = [
   "Ralph Giles <giles@mozilla.com>",
   "Matthew Gregan <kinetik@flim.org>",
+  "Guillaume Gomez <guillaume1.gomez@gmail.com>"
```
Comment (rillian) [inline Cargo.toml]: If you don't mind I think it makes more sense to keep the author's list to just the two primary contacts. Your contributions will be clear in the commit history.

**Comment by rillian** [comment]
Thanks, that's better. One more comment to address and then I'm happy to merge.

File: Cargo.toml (hunk 6 above)
Comment (GuillaumeGomez) [inline Cargo.toml]: Fine by me! I remove it.

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by GuillaumeGomez** [comment]
When do you think this version will be published?

**Comment by rillian** [comment]
I was hoping to get the other PRs merged first, but I can publish what we have now if that's helpful.

**Comment by rillian** [comment]
I don't know if making more symbols public counts as a minor or patch update under semver, but we're on 0.x so it doesn't matter! I've published https://crates.io/crates/mp4parse/0.4.1 with your change. Hope that's enough to get your code working.

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**